presto-main/src/main/java/io/prestosql/sql/planner/plan/AggregationNode.java

```diff
@@ -253,7 +253,9 @@ public boolean hasSingleNodeExecutionPreference(Metadata metadata)
 
     public boolean isStreamable()
     {
-        return !preGroupedSymbols.isEmpty() && groupingSets.getGroupingSetCount() == 1 && groupingSets.getGlobalGroupingSets().isEmpty();
+        return ImmutableSet.copyOf(preGroupedSymbols).equals(ImmutableSet.copyOf(groupingSets.getGroupingKeys()))
```

Comment:
This is not the same check as before. If pre-grouped symbols is a subset of the grouping keys, the old check would return `true` (`!preGroupedSymbols.isEmpty()`). In the new check, they have to match exactly the grouping keys.

Reply:
> This is not the same check as before.

The previous check was wrong. `StreamingAggregationOperator` is used when `isStreamable()` returns `true`. However, this means that `preGrouped` symbols might be subset of group-by symbols. `StreamingAggregationOperator` doesn't support that as it doesn't store any pages so it cannot group remaining symbols.

Luckly, the code in `AddLocalExchanges.Rewriter#visitAggregation` (which adds `preGruoped` symbols to `AggregationNode`) adds pre-grouped symbols field only when it matches `groupingKeys`.